### PR TITLE
[13.0][MIG] base_business_document_import

### PR DESCRIPTION
**Pull request by sebalix**

### CODE REVIEW

File: base_business_document_import/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "category": "Tools",
+    "license": "AGPL-3",
+    "summary": "Provides technical tools to import sale orders or supplier invoices",
+    "author": "Akretion, Nicolas JEUDY, Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/edi",
+    "depends": [
+        # odoo
+        "account",
+        # OCA/partner-contact
+        "base_vat_sanitized",
```
Comment (astirpe) [inline base_business_document_import/__manifest__.py]: Please remove `base_vat_sanitized` from the dependencies.
See also https://github.com/OCA/partner-contact/pull/821#issuecomment-593145213

and

https://github.com/OCA/partner-contact/blob/13.0/base_vat_sanitized/readme/DESCRIPTION.rst

File: base_business_document_import/models/business_document_import.py [hunk 2]
```diff
@@ -0,0 +1,1100 @@
...
+                ],
+                limit=1,
+            )
+            if state:
+                domain += ["|", ("state_id", "=", False), ("state_id", "=", state.id)]
+        if partner_dict.get("vat"):
+            vat = partner_dict["vat"].replace(" ", "").upper()
+            # use base_vat_sanitized
+            partner = rpo.search(
+                domain + [("parent_id", "=", False), ("sanitized_vat", "=", vat)],
```
Comment (astirpe) [inline base_business_document_import/models/business_document_import.py]: Please adapt this part to use field `vat` of standard Odoo

File: base_business_document_import/models/business_document_import.py [hunk 3]
```diff
@@ -0,0 +1,1100 @@
...
+                    ("code", "=", partner_dict["state_code"]),
+                    ("country_id", "=", country.id),
+                ],
+                limit=1,
+            )
+            if state:
+                domain += ["|", ("state_id", "=", False), ("state_id", "=", state.id)]
+        if partner_dict.get("vat"):
+            vat = partner_dict["vat"].replace(" ", "").upper()
+            # use base_vat_sanitized
```
Comment (astirpe) [inline base_business_document_import/models/business_document_import.py]: `base_vat_sanitized` should be not used

**Review by astirpe — CHANGES_REQUESTED** [review]

File: base_business_document_import/__manifest__.py (hunk 1 above)
Comment (sebalix) [reply base_business_document_import/__manifest__.py]: Thanks, the dependency has been removed and the code updated

File: base_business_document_import/models/business_document_import.py [hunk 4]
```diff
@@ -1020,7 +1019,7 @@ def _check_company(self, company_dict, chatter_msg):
         if company_dict.get("vat"):
             parsed_company_vat = company_dict["vat"].replace(" ", "").upper()
             if company.partner_id.sanitized_vat:
```
Comment (astirpe) [inline base_business_document_import/models/business_document_import.py]:
```suggestion
            if company.partner_id.vat:
```

Comment (astirpe) [reply base_business_document_import/models/business_document_import.py]: Sorry you were faster :smile:

**Comment by sebalix** [comment]
@astirpe PR green, ready to merge, unless you find something else :)

**Review by astirpe — APPROVED** [review]
Code review only  (I don't have time to test it) :+1:

**Review by i-vyshnevska — APPROVED** [review]
LG

File: base_business_document_import/tests/test_business_document_import.py [hunk 5]
```diff
@@ -9,13 +9,7 @@
...
     def test_match_partner(self):
         partner1 = self.env["res.partner"].create(
-            {
-                "name": "Total SA",
-                "supplier": False,
-                "customer": False,
-                "ref": "TOTAL",
-                "website": "www.total.com",
-            }
+            {"name": "Total SA", "ref": "TOTAL", "website": "www.total.com"}
```
Comment (Tonow-c2c) [inline base_business_document_import/tests/test_business_document_import.py]: Prefer a fake company :wink: :fuelpump:

Comment (sebalix) [reply base_business_document_import/tests/test_business_document_import.py]: You're right, let's be less opinionated :)

File: base_business_document_import/tests/test_business_document_import.py [hunk 6]
```diff
@@ -9,11 +9,11 @@
 class TestBaseBusinessDocumentImport(TransactionCase):
     def test_match_partner(self):
         partner1 = self.env["res.partner"].create(
-            {"name": "Total SA", "ref": "TOTAL", "website": "www.total.com"}
+            {"name": "COGIP", "ref": "COGIP", "website": "http://example.com/"}
```
Comment (Tonow-c2c) [inline base_business_document_import/tests/test_business_document_import.py:12]: :+1: :tada:

**Review by Tonow-c2c — APPROVED** [review]

**Review by jgrandguillaume — APPROVED** [review]
LGTM

**Comment by pedrobaeza** [comment]
There has been a problem with merge bot, but it's fixed now, so you can relaunch.

**Comment by pedrobaeza** [comment]
ocabot stuck. Please relaunch

**Comment by jgrandguillaume** [comment]
@pedrobaeza Thanks for checking this

**Pull request merged**